resources/2024-10-dapr-bindings.md

```diff
@@ -0,0 +1,368 @@
+# Dapr Binding Implementation
+
+* **Author**: Lucas Peirone (@SoTrx)
+
+## Overview
+
+This design document proposes the addition of support for Dapr Bindings within Radius. By integrating Dapr Bindings, resources will be able to interact with external systems in a decoupled and scalable manner, facilitating better communication between services and external resources.
+
+## Terms and definitions
+
+[Dapr](https://github.com/dapr/dapr): Distributed Application Runtime. It provides "building blocks" for writing microservices, offering high-level functionalities for common service tasks.
+
+**External Resource**: A resource that is not part of the application's infrastructure. This could be a database, queue, storage system, etc. It may be a service created by another team within the same organization, a third-party service, or a service created by a different organization. No management rights over the external resource are assumed.
+
+[**Dapr Bindings**](https://docs.dapr.io/developing-applications/building-blocks/bindings/bindings-overview/): A Dapr building block that enables interaction with external resources in a loosely coupled way. Bindings are used to interact with resources like databases, queues, and storage systems.
+
+Bindings are further categorized into two types: input and output bindings:
+- **Input bindings** are used to receive events from an external resource. [Example](https://docs.dapr.io/reference/components-reference/supported-bindings/kafka/)
+- **Output bindings** are used to send events to an external resource. [Example](https://docs.dapr.io/reference/components-reference/supported-bindings/smtp/)
+- These two categories are not mutually exclusive. Some bindings can serve as both an input and output binding.
+
+For a single Dapr Binding, the following resources are defined:
+- **Exactly one** Dapr component: This is a [Kubernetes Custom Resource Definition (CRD)](https://docs.dapr.io/reference/api/bindings_api/) that defines the binding's configuration, including authentication information.
+- **Zero or one** External Resource: In some cases, the binding does not require an external resource, making it self-sufficient. An example of this is the [CRON binding](https://docs.dapr.io/reference/components-reference/supported-bindings/cron/).
+
+
+
+## Objectives
+
+> **Issue Reference:** radius-project/radius#7960
+
+### Goals
+
+- Enable users to create, update, and delete Dapr Bindings Component (CRD) using Radius.
+- Create a list of recipes covering common use cases for Dapr Bindings. 
+
+### Non goals
+
+- (out-of-scope) Managing **external resources** used by Dapr Bindings. As we do not have management rights over the external resources, we will not be able to create, update, or delete them. The user will need to create the external resource and provide the necessary information to the Dapr Binding.
+
+### User scenarios (optional)
```

Comment:
In both user scenarios, I recommend elaborating on the user ultimately leveraging Dapr Bindings to accomplish the abstraction, and then they are wanting to make use of these Dapr Bindings in Radius applications. e.g.
> As an operator using Radius, I want the application I manage to interface with other applications in my organization without exposing their implementation details to the developers by leveraging Dapr Bindings. Thus, I would like to define and use Dapr Bindings as a resource type within my Radius applications.